core/types/pow/cuckaroo.go

```diff
@@ -43,7 +43,7 @@ func (this *Cuckaroo) Verify(headerData []byte, blockHash hash.Hash, targetDiffB
 			"less than min diff :%d", targetDiff, this.params.CuckarooMinDifficulty)
 		return errors.New(str)
 	}
-	if CalcCuckooDiff(GraphWeight(uint32(edgeBits),this.mainHeight,CUCKAROO), blockHash).Cmp(targetDiff) < 0 {
+	if CalcCuckooDiff(GraphWeight(uint32(edgeBits),this.mainHeight,this.params.AdjustmentStartMainHeight,CUCKAROO), blockHash).Cmp(targetDiff) < 0 {
```

Comment:
the `AdjustmentStartMainHeight` param should not be used intrusively.

Reply:
`AdjustmentStartMainHeight` is part of consensus but `GraphWeight` function should not be part of the consensus. it should remain a utility function to keep out of aware of the consensus